src/main/scala/no/ndla/audioapi/service/ConverterService.scala

```diff
@@ -8,51 +8,53 @@
 
 package no.ndla.audioapi.service
 
-import com.netaporter.uri.Uri
 import com.typesafe.scalalogging.LazyLogging
-import com.netaporter.uri.dsl._
 import no.ndla.audioapi.AudioApiProperties._
 import no.ndla.audioapi.auth.User
-import no.ndla.audioapi.model.api.NotFoundException
-import no.ndla.audioapi.model.domain.{Audio, Tag, Title}
-import no.ndla.audioapi.model.Language._
+import no.ndla.audioapi.model.Language.{findByLanguageOrBestEffort, DefaultLanguage}
 import no.ndla.audioapi.model.{api, domain}
 import no.ndla.mapping.License.getLicense
 
-import scala.util.{Failure, Success, Try}
+import scala.util.{Success, Try}
 
 
 trait ConverterService {
   this: User with Clock =>
   val converterService: ConverterService
 
   class ConverterService extends LazyLogging {
-    def toApiAudioMetaInformation(audioMeta: domain.AudioMetaInformation, language: String): Try[api.AudioMetaInformation] = {
-      val lang = language match {
-        case AllLanguages if audioMeta.supportedLanguages.contains(DefaultLanguage) => DefaultLanguage
-        case AllLanguages if audioMeta.supportedLanguages.nonEmpty => audioMeta.supportedLanguages.head
-        case l => l
-      }
-
-      if (!audioMeta.supportedLanguages.contains(lang))
-        return Failure(new NotFoundException)
 
-      val audioFile = findByLanguage(audioMeta.filePaths, lang).getOrElse(audioMeta.filePaths.head)
+    def toApiAudioMetaInformation(audioMeta: domain.AudioMetaInformation, language: Option[String]): Try[api.AudioMetaInformation] = {
       Success(api.AudioMetaInformation(
         audioMeta.id.get,
         audioMeta.revision.get,
-        lang,
-        findByLanguage(audioMeta.titles, lang).getOrElse(""),
-        toApiAudio(audioFile),
+        toApiTitle(findByLanguageOrBestEffort[String, domain.Title](audioMeta.titles, language)),
```

Comment:
Nettopp :-) 

Akkurat den der er jeg meget misfornøyd med selv, og jeg digger at vi har pull-requests, for når jeg nå var på vei til å forklare hvorfor den må være som den er, så fant jeg løsningen som kan forenkle dette :-)

Så, signaturen var 
```
def findByLanguageOrBestEffort[T, P <: LanguageField[T]](sequence: Seq[P], lang: Option[String]): Option[P] = {
```  

Der `T` representerer datatypen på verdien som er språkstyrt og der `P` angir datatypen på wrapperen.
F.eks for en case class som ser noe a'la det her ut `case class Title(title: String, language: String)` så vil `T = String` og `P = Title`. 

Den klarte ikke å utlede fra input-parametrene hva `T` og `P` var, det var derfor måtte vi angi det i metodekallet `findByLanguageOrBestEffort[String, domain.Title](titles, language)`. 

Men, ved å ikke bry oss om hva datatypen på selve verdien er, så klarer den å utlede det selv.
Så ved å endre signaturen til 
```
def findByLanguageOrBestEffort[P <: LanguageField[_]](sequence: Seq[P], lang: Option[String]): Option[P] = {
```
så kan vi endre kallet til denne metoden til å være slik vi egentlig vil ha det, som er `findByLanguageOrBestEffort(titles, language)`